packages/svelte/src/internal/client/validate.js

```diff
@@ -85,13 +85,20 @@ export function loop_guard(timeout) {
 /**
  * @param {Record<string, any>} $$props
  * @param {string[]} bindable
+ * @param {string[]} exports
  */
-export function validate_prop_bindings($$props, bindable) {
+export function validate_prop_bindings($$props, bindable, exports) {
 	for (const key in $$props) {
-		if (!bindable.includes(key)) {
-			var setter = get_descriptor($$props, key)?.set;
+		var setter = get_descriptor($$props, key)?.set;
 
-			if (setter) {
+		if (setter) {
+			if (exports.includes(key)) {
+				throw new Error(
+					`Cannot use bind:${key} on this component because it is a component export, and you can only bind to properties in runes mode. ` +
+						`Use bind:this instead and then access the property on the bound component instance.`
```

Comment:
This also applies to the existing error message for non-bindable props, but: the message is a bit cryptic because it refers to 'this component' with no further context. 'you can only bind to properties in runes mode' is also slightly cryptic because the parent component is the one doing the binding, so it sounds like it matters which mode the parent is in rather than the child.

Ideally we would specify both parent and child component. Something like

> Cannot use `bind:foo` (in Parent.svelte) to bind to an export from Child.svelte. Instead, use `bind:this={component}` and access the property as `component.foo`

and

> Cannot use `bind:foo` (in Parent.svelte) to bind to a non-bindable prop. To make it bindable, use the `$bindable` rune in Child.svelte: `let { foo = $bindable } = $props()`

Reply:
Tweaked the messages, let me know if it's clear enough now or if we need to find a way to bring the consumer component name in there.